### PR TITLE
Update to gradle 7

### PR DESCRIPTION
**Pull request by dsvoronin**
https://docs.gradle.org/7.0/release-notes.html

## Benchmark

see `performance.scenarios` for options

| gradle version  | fastCheck median |
| ------------- | ------------- |
| 6.9 | 24,621.41 ms |
| 7.0.2  | 24,955.83 ms |

## TODO

- [X] check kotlin daemon params gradle/gradle#17075 potential blocker; should check before upgrade (FIXED in 7.2) Only blocks avito update
- [x] https://github.com/avito-tech/avito-android/pull/1050 blocker

### CODE REVIEW

**Comment by dsvoronin** [comment]
https://github.com/gradle/gradle/issues/17075 potential blocker; should check before upgrade

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.8.3-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.0-all.zip
```
Comment (eugene-krivobokov) [inline gradle/wrapper/gradle-wrapper.properties]: https://github.com/gradle/gradle/releases/tag/v7.0.1

Comment (eugene-krivobokov) [reply gradle/wrapper/gradle-wrapper.properties]: https://github.com/gradle/gradle/releases/tag/v7.0.2

**Comment by dsvoronin** [comment]
> [gradle/gradle#17075](https://github.com/gradle/gradle/issues/17075) potential blocker; should check before upgrade

We are blocked with 7.2 then https://github.com/gradle/gradle/pull/17432

File: subprojects/gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.9-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.0.2-all.zip
```
Comment (RuslanMingaliev) [inline subprojects/gradle/wrapper/gradle-wrapper.properties]: 7.1?

File: subprojects/settings.gradle.kts [hunk 3]
```diff
@@ -169,26 +169,14 @@ pluginManagement {
                 }
             }
             filter {
-                includeGroupByRegex("com\\.android\\.tools\\.build\\.*")
+                includeGroupByRegex("com\\.android.*")
+                includeGroupByRegex("androidx.*")
+                includeGroup("com.google.test.platform") // todo remove after upgrade to 4.2.x+
```
Comment (RuslanMingaliev) [inline subprojects/settings.gradle.kts]: we have 4.2.1

File: subprojects/gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.9-all.zip
```
Comment (eugene-krivobokov) [inline subprojects/gradle/wrapper/gradle-wrapper.properties:3]: Why not 7.1?

**Review by eugene-krivobokov — APPROVED** [review]

**Review by RuslanMingaliev — APPROVED** [review]

File: subprojects/gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (dsvoronin) [reply subprojects/gradle/wrapper/gradle-wrapper.properties]: There was a problem with change Copy task behavior;
If multiple `into` specified, only first one actually copied

Comment (eugene-krivobokov) [reply subprojects/gradle/wrapper/gradle-wrapper.properties]: If I'm not mistaken, this behavior was already in 6.9.

**Pull request merged**